packages/arb-token-bridge-ui/src/util/NetworkUtils.ts

```diff
@@ -0,0 +1,114 @@
+import { hexValue } from 'ethers/lib/utils'
+import { ChainId, getNetworkName, rpcURLs, getExplorerUrl } from './networks'
+import { BigNumber } from 'ethers'
+import { Web3Provider } from '@ethersproject/providers'
+import EthereumLogo from '../assets/EthereumLogo.png'
+import ArbitrumOneLogo from '../assets/ArbitrumOneLogo.svg'
+import ArbitrumNovaLogo from '../assets/ArbitrumNovaLogo.png'
+
+/* 
+  The list which will be available for network selection in navbar-dropdowns plus
+  Icons and colors attached with our networks available for selection
+*/
+export const supportedNetworks: {
+  [key in ChainId]?: { img: string; btnThemeClass: string }
+} = {
+  [ChainId.Mainnet]: {
+    img: EthereumLogo,
+    btnThemeClass: 'bg-blue-arbitrum'
+  },
+  [ChainId.ArbitrumOne]: {
+    img: ArbitrumOneLogo,
+    btnThemeClass: 'bg-blue-arbitrum'
+  },
+  [ChainId.ArbitrumNova]: {
+    img: ArbitrumNovaLogo,
+    btnThemeClass: 'bg-brick-dark'
+  }
+}
+
+export type SwitchChainProps = {
+  chainId: ChainId
+  provider: Web3Provider
+  onSuccess?: () => void
+  onError?: (err?: Error) => void
+  onSwitchChainNotSupported?: () => void
+}
+
+const noop = () => {}
+
+const isSwitchChainSupported = (provider: Web3Provider) =>
+  provider?.provider &&
+  //@ts-ignore
+  (provider.provider.isMetaMask || provider.provider.isImToken)
```

Comment:
If you check the old implementation, `Web3Provider` was a custom type:

https://github.com/OffchainLabs/arb-token-bridge/blob/ed13c6ce7e34cca0926fa559ee2b8ffd29d29a28/packages/arb-token-bridge-ui/src/components/App/App.tsx#L66